### ISSUE TITLE
Adgaurd Limit Concurrent requests per second

### ISSUE THREAD

**Issue by miladtempaccount** [AdguardTeam/AdGuardHome#7463]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, AMD64 (aka x86_64)

### Installation

GitHub releases or script from README

### Setup

On one machine

### AdGuard Home version

latest

### Action

Just benchmark the adguard and you will see it won't go higher than around 27K Query Per Second

```root@cr-srv-st:~/test1# dnsperf -s 127.0.0.1 -p 53561 -c 1 -d test/query.txt -n 1
DNS Performance Testing Tool
Version 2.14.0

[Status] Command line: dnsperf -s 127.0.0.1 -p 53561 -c 1 -d test/query.txt -n 1
[Status] Sending queries (to 127.0.0.1:53561)
[Status] Started at: Tue Nov 26 06:11:41 2024
[Status] Stopping after 1 run through file
[Status] Testing complete (end of file)

Statistics:

  Queries sent:         7020
  Queries completed:    7020 (100.00%)
  Queries lost:         0 (0.00%)

  Response codes:       NOERROR 7020 (100.00%)
  Average packet size:  request 27, response 43
  Run time (s):         0.284940
  Queries per second:   24636.765635

  Average Latency (s):  0.003639 (min 0.000275, max 0.013657)
  Latency StdDev (s):   0.001711


```

### Expected result

around 50K or 100K per second at least

### Actual result

around 26K per second

### Additional information and/or screenshots

There is no There is no hardware limit, all good all ok, Please tell me what can I do ? 

I even set max goroutine to 1500, but no difference 

Please tell me what can I do ? 

here is test results of my own dns server ( upstream of adguard ) 

`````
root@cr-srv-st:~/test1# dnsperf -s 127.0.0.1 -p 5353 -c 1 -d test/query.txt -n 10
DNS Performance Testing Tool
Version 2.14.0

[Status] Command line: dnsperf -s 127.0.0.1 -p 5353 -c 1 -d test/query.txt -n 10
[Status] Sending queries (to 127.0.0.1:5353)
[Status] Started at: Tue Nov 26 06:13:31 2024
[Status] Stopping after 10 runs through file
[Status] Testing complete (end of file)

Statistics:

  Queries sent:         70200
  Queries completed:    70200 (100.00%)
  Queries lost:         0 (0.00%)

  Response codes:       NOERROR 70200 (100.00%)
  Average packet size:  request 27, response 63
  Run time (s):         1.050964
  Queries per second:   66795.817935

  Average Latency (s):  0.001348 (min 0.000051, max 0.014170)
  Latency StdDev (s):   0.000915


`````

**Comment by miladtempaccount** [comment]
./dnsproxy -l 127.0.0.1 -p 5355 -u 127.0.0.1:5353 --max-go-routines=1000000


root@cr-srv-st:~/test1# dnsperf -s 127.0.0.1 -p 5355 -c 1 -d test/query.txt -n 10
DNS Performance Testing Tool
Version 2.14.0

[Status] Command line: dnsperf -s 127.0.0.1 -p 5355 -c 1 -d test/query.txt -n 10
[Status] Sending queries (to 127.0.0.1:5355)
[Status] Started at: Tue Nov 26 08:51:25 2024
[Status] Stopping after 10 runs through file
[Status] Testing complete (end of file)

Statistics:

  Queries sent:         70200
  Queries completed:    70200 (100.00%)
  Queries lost:         0 (0.00%)

  Response codes:       NOERROR 70200 (100.00%)
  Average packet size:  request 27, response 43
  Run time (s):         2.805360
  Queries per second:   25023.526392

  Average Latency (s):  0.003816 (min 0.000183, max 0.028328)
  Latency StdDev (s):   0.002333

**Comment by vincejv** [comment]
how does dnsmasq, powerdns and unbound compare?

**Comment by miladtempaccount** [comment]
I get over 70K request per second on my self developed dns server also I get much more on dnsmasq and unbount, please tell me what can I do to increase the speed 

It's not using hardware resources, I have enough cpu and ram

**Comment by sofvaskina** [comment]
Yes, you're right. The problem is with dnsproxy. If you need more than 50K requests per second, you can try running multiple instances behind a load balancer. But generally speaking, 26K requests per second is ~2 billion requests per day - which seems more than enough for any practical deployment.
